plugin/core/sessions.py

```diff
@@ -1489,7 +1489,7 @@ def _handle_initialize_success(self, result: InitializeResult) -> None:
             self._workspace_folders = self._workspace_folders[:1]
         self.state = ClientStates.READY
         if self._plugin_class is not None:
-            self._plugin = self._plugin_class(weakref.ref(self))  # type: ignore  # TODO fix me
+            self._plugin = self._plugin_class(weakref.ref(cast(Session, self)))
```

Comment:
This type issue is also a bug:

https://github.com/microsoft/pyright/discussions/6334
https://github.com/microsoft/pyright/issues/6335

Reply:
So we should rather wait for the release of an unbugged Pyright version, I guess?

Reply:
yeah